node/flatpak_node_generator/providers/yarn.py

```diff
@@ -37,88 +39,72 @@ def is_git_version(version: str) -> bool:
             return len([p for p in url.path.split('/') if p]) == 2
         return False
 
+    def parse_lockfile_v1(self, lockfile: Path) -> Dict[str, Any]:
+        def _iter_lines() -> Iterator[Tuple[int, str]]:
+            indent = '  '
+            for line in lockfile.open('r'):
+                level = 0
+                while line.startswith(indent):
+                    level += 1
+                    line = line[len(indent) :]
+                yield level, line.strip()
+
+        root_entry: Dict[str, Any] = OrderedDict()
+        parent_entries = [root_entry]
+
+        for level, line in _iter_lines():
+            if line.startswith('#') or not line:
+                continue
+            assert level <= len(parent_entries) - 1
+            parent_entries = parent_entries[: level + 1]
+            if line.endswith(':'):
+                child_entry = parent_entries[-1][line[:-1]] = OrderedDict()
+                parent_entries.append(child_entry)
+            else:
+                # NOTE shlex.split is handy, but slow;
+                # to speed up parsing we can use something less robust, e.g.
+                # _key, _value = line.split(' ', 1)
+                # parent_entries[-1][self.unquote(_key)] = self.unquote(_value)
+                _key, _value = shlex.split(line)
+                parent_entries[-1][_key] = _value
+
+        return root_entry
+
     def unquote(self, string: str) -> str:
```

Comment:
Now that shlex is used (correctly! I should've used that in the first place, really) is this still needed?

Reply:
oops I realized that this was used later and thought I deleted this comment...but apparently didn't :sweat_smile: please ignore